src/plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts

```diff
@@ -1,112 +1,105 @@
-import { Reviewer, ReviewAssignmentObject, AssignmentStatus } from './types'
+import { Reviewer, ReviewAssignmentObject, AssignmentStatus, ApplicationData } from './types'
 import databaseMethods from './databaseMethods'
 
 module.exports['generateReviewAssignments'] = async function (input: any, DBConnect: any) {
   const db = databaseMethods(DBConnect)
 
   console.log('Generating review assignment records...')
   try {
-    const { applicationId, reviewId, templateId, stageId, stageNumber } = input
-    // NB: reviewId comes from record_id on TriggerPayload when triggered
-    // from review table
+    const { applicationId, reviewId } = input
+    // Get template information and current stage for application
+    const applicationData: ApplicationData = await DBConnect.getApplicationData(applicationId)
```

Comment:
No need to query this, it's already passed through as part of input. So you just need to add `applicationData` to the destructured field above:
`const { applicationId, reviewId, applicationData } = input`

Reply:
Nice!